checker/check_result.go

```diff
@@ -94,24 +94,24 @@ func CreateProportionalScore(b, t int) int {
 // and normalizes the result.
 // Each score contributes equally.
 func AggregateScores(scores ...int) int {
-	n := len(scores)
-	r := float32(0)
+	n := float64(len(scores))
+	r := 0
 	for _, s := range scores {
-		r += float32(s) / float32(MaxResultScore) * float32(n)
+		r += s
 	}
-	return int(r)
+	return int(math.Floor(float64(r) / n))
 }
 
 // AggregateScoresWithWeight adds up all scores
 // and normalizes the result.
 // The caller is responsible for ensuring the sum of
 // weights is 10.
 func AggregateScoresWithWeight(scores map[int]int) int {
-	r := float32(0)
+	r := 0
 	for s, w := range scores {
-		r += float32(s) / float32(MaxResultScore) * float32(w)
+		r += s * w
 	}
-	return int(r)
+	return int(math.Floor(float64(r) / float64(MaxResultScore)))
```

Comment:
> Any reason why we require callers to make sure the weights add up to 10, rather than summing them ourselves and using that as the denominator?
+1. implemented.